gui/src/app/SPAnalysis/SPAnalysisDataModel.ts

```diff
@@ -9,28 +9,66 @@ type SPAnalysisFiles = {
     [filetype in SPAnalysisKnownFiles]: string
 }
 
+const isSPAnalysisFiles = (x: any): x is SPAnalysisFiles => {
+    if (!x) return false
+    if (typeof x !== 'object') return false
+    for (const k of Object.values(SPAnalysisKnownFiles)) {
+        if (typeof x[k] !== 'string') return false
```

Comment:
Does this also do the right thing if x does not have one of the keys? I can’t recall if object indexing with an invalid key returns undefined or throws

Reply:
In that case `x[k]` would be undefined and `typeof x[k] !== 'string'` would be true, so it would return false.

Perhaps we should allow it to be undefined? Certainly in the future we will not want to require data.py, etc. Should we mark files as optional when we get to that point?

Reply:
If we see a reason to differentiate between “undefined” and “defined but empty” we could do that when the time comes, but I think internally an empty string would be a fine representation for a missing file